solrsearch-plugins/src/main/java/co.cask.hydrator.plugin/batch/SolrSearchSink.java

```diff
@@ -0,0 +1,216 @@
+    uniqueKey = config.getIdField();
+    outputFieldMappings = config.getOutputFieldMappings();
+    solrClient = getSolrConnection();
+  }
+
+  @Override
+  public void transform(StructuredRecord structuredRecord, Emitter<KeyValue<SolrInputField, SolrInputDocument>> emitter)
+    throws Exception {
+    String solrFieldName;
+    SolrInputDocument document;
+
+    validateIdField(structuredRecord.getSchema());
+    validateInputFieldsDataType(structuredRecord.getSchema());
+
+    if (structuredRecord.get(uniqueKey) == null) {
+      return;
+    }
+    document = new SolrInputDocument();
+    for (Schema.Field field : structuredRecord.getSchema().getFields()) {
+      solrFieldName = field.getName();
+      if (outputFieldMappings.containsKey(solrFieldName)) {
+        document.addField(outputFieldMappings.get(solrFieldName), structuredRecord.get(solrFieldName));
+      } else {
+        document.addField(solrFieldName, structuredRecord.get(solrFieldName));
+      }
+    }
+    solrClient.add(document);
+    solrClient.commit();
+    emitter.emit(new KeyValue<>(document.getField(uniqueKey), document));
+  }
+
+  @Override
+  public void destroy() {
+    solrClient.shutdown();
+  }
+
+  @Override
+  public void onRunFinish(boolean succeeded, BatchSinkContext context) {
+    File directory = new File(tempDir.getAbsolutePath() + "/temp");
+    if (directory.exists()) {
+      directory.delete();
+    }
+  }
+
+  /**
+   * Returns the SolrClient for establishing the connection with Solr server, using the hostname and port provided by
+   * user.
+   *
+   * @return Solr client
+   */
+  private SolrClient getSolrConnection() {
+    String urlString;
+    SolrClient solrClient = null;
+    if (config.getSolrMode().equals(SINGLE_NODE_SOLR_MODE)) {
+      urlString = "http://" + config.getSolrHost() + "/solr/" + config.getCollectionName();
+      solrClient = new HttpSolrClient(urlString);
+    } else if (config.getSolrMode().equals(SOLR_CLOUD_MODE)) {
+      CloudSolrClient solrCloudClient = new CloudSolrClient(config.getSolrHost());
+      solrCloudClient.setDefaultCollection(config.getCollectionName());
+      solrClient = solrCloudClient;
+    }
+    return solrClient;
+  }
+
+  /**
+   * Validates whether the host entered for Single Node Solr instance is proper or not.
+   */
+  private void validateSolrConnectionString() {
+    if (config.getSolrMode().equals(SINGLE_NODE_SOLR_MODE) && config.getSolrHost().contains(",")) {
+      throw new IllegalArgumentException(String.format("Multiple hosts '%s' found for Single Node Solr.",
+                                                       config.getSolrHost()));
+    }
+  }
+
+  /**
+   * Verifies whether the Solr server is running or not. Also, validates whether the collection provided by user is
+   * registered with the Solr server or not.
+   */
+  private void verifySolrConfiguration() {
+    SolrClient client = getSolrConnection();
+    try {
+      client.ping();
```

Comment:
is it possible to ping without specifying a collection?
That way, we can know whether the issue is the host/port or the collection not existing?